dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/TeradataTablesValidatorTask.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2022-2024 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.teradata;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.ByteSink;
+import com.google.edwmigration.dumper.application.dumper.MetadataDumperUsageException;
+import com.google.edwmigration.dumper.application.dumper.handle.JdbcHandle;
+import com.google.edwmigration.dumper.application.dumper.task.AbstractJdbcTask;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Validate if the provided tables exist/are accessible for the task and can be used later in the
+ * tasks flow
+ */
+public class TeradataTablesValidatorTask extends AbstractJdbcTask<Void> {
+  private static final Logger LOG = LoggerFactory.getLogger(TeradataTablesValidatorTask.class);
+
+  private final ImmutableList<String> tableNames;
+
+  public TeradataTablesValidatorTask(@Nonnull String... tableNames) {
+    super(TeradataTablesValidatorTask.class.getSimpleName() + ".txt", false);
+    Preconditions.checkNotNull(tableNames, "Validated table names are null");
```

Comment:
You can use static import to remove boilerplate.

Reply:
is it widely accepted style guide which we should follow?  I personally prefer to keep `Preconditions` for readability.

Reply:
"Static imports are appropriate when the usage is clear without additional context of the enclosing type. Static imports serve to remove redundant context and reduce boilerplate. The method/field should be self-describing or fairly well-known as a static import."

Reply:
I see that in other places static import is not used, so it's better to stay consistent (even if it's boilerplate code), so I take back my recommendation.